src/Amazon.Lambda.Tools/Commands/UpdateFunctionConfigCommand.cs

```diff
@@ -248,7 +248,13 @@ protected async Task UpdateConfigAsync(GetFunctionConfigurationResponse existing
             }
 
             var urlConfig = await this.GetFunctionUrlConfig(existingConfiguration.FunctionName);
-            var enableUrlConfig = this.GetBoolValueOrDefault(this.FunctionUrlEnable, LambdaDefinedCommandOptions.ARGUMENT_FUNCTION_URL_ENABLE, false).GetValueOrDefault();
+
+            // To determine what is the state of the function url check to see if the user explicitly set a value. If they did set a value then use that 
+            // to either add or remove the url config. If the user didn't set a value check to see if there is an existing config to make sure we don't remove
+            // the config url if the user didn't set a value.
+            bool enableUrlConfig = this.GetBoolValueOrDefault(this.FunctionUrlEnable, LambdaDefinedCommandOptions.ARGUMENT_FUNCTION_URL_ENABLE, false).HasValue ? 
+                                        this.GetBoolValueOrDefault(this.FunctionUrlEnable, LambdaDefinedCommandOptions.ARGUMENT_FUNCTION_URL_ENABLE, false).Value : urlConfig != null;
```

Comment:
```suggestion
            var functionUrlEnableArg = this.GetBoolValueOrDefault(this.FunctionUrlEnable, LambdaDefinedCommandOptions.ARGUMENT_FUNCTION_URL_ENABLE, false);
            bool enableUrlConfig = functionUrlEnableArg.HasValue ? functionUrlEnableArg.Value : urlConfig != null;
```

Reply:
That is how I had it originally but my concern was having 2 variables `functionUrlEnableArg (bool?)` and `enableUrlConfig (bool)` with such similar names and basically the same datatype would get miss used in the future.